gflownet/envs/tree.py

```diff
@@ -824,17 +824,21 @@ def state2policy_mlp(
         """
         if state is None:
             state = self.state.clone().detach()
-        state[state.isnan()] = -2
-        return state.flatten()
+        return self.statetorch2policy_mlp(state.unsqueeze(0))[0]
 
     def statetorch2policy_mlp(
         self, states: TensorType["batch_size", "state_dim"]
     ) -> TensorType["batch_size", "policy_input_dim"]:
         """
         Prepares a batch of states in torch "GFlowNet format" for an MLP policy model.
-        It simply replaces the NaNs by -1s.
+        It replaces the NaNs by -2s, removes the activity attribute, and explicitly
+        appends the attribute vector of the active node (if present).
         """
+        m, n = torch.where(states[:, :-1, Attribute.ACTIVE] == Status.ACTIVE)
+        active_features = torch.full((states.shape[0], 1, 4), -2.0)
+        active_features[m] = states[m, n, : Attribute.ACTIVE].unsqueeze(1)
```

Comment:
`m` and `n` are rows and columns right?

Minor suggestion: if that is the case, I would use `rows` and `cols` instead just to make it a bit easier to understand.

Reply:
Yes, modified as suggested.